code/game/objects/structures/mirror.dm

```diff
@@ -54,12 +61,51 @@
 /obj/structure/mirror/obj_break(damage_flag, mapload)
 	if(!broken && !(flags_1 & NODECONSTRUCT_1))
 		icon_state = "mirror_broke"
+		vis_contents -= reflection
 		if(!mapload)
 			playsound(src, "shatter", 70, TRUE)
-		if(desc == initial(desc))
-			desc = "Oh no, seven years of bad luck!"
+		desc = "Oh no, seven years of bad luck!"
 		broken = TRUE
 
+/**
+  * If a (living, non-ghost) mob is within 2 tiles of the mirror, their reflection is displayed.
+  *
+  * The mob is added to the mirror container's vis_contents, which is scaled down, reflected, and alpha masked.
+  * The mirror container in turn has the reflection in its vis_contents.
+  */
+
+/obj/structure/mirror/HasProximity(atom/movable/AM as mob|obj)
+	if(!isliving(AM) || broken)
+		return
+	RegisterSignal(AM, COMSIG_MOVABLE_MOVED, .proc/check_reflection, override = TRUE)
+	reflection.vis_contents += AM
+	reflection.filters += filter(type="alpha", icon = icon(src.icon, "mirror_mask"))
+	var/matrix/M = matrix()
+	M.Scale(0.75)
+	M.Multiply(matrix(-1, 0, 0, 0, 1, 0)) // reflection matrix
+	reflection.transform = M
+	vis_contents += reflection
+	if(AM.GetComponent(/datum/component/mood) && AM.GetComponent(/datum/component/mood).sanity <= SANITY_DISTURBED)
+		desc = "Despite everything, it's still you."

```

Comment:
but what if someone else is sntaind gcloser to the mirror and its reflectign them so its actually not you

Reply:
Also, use a signal instead of `GetComponent`